src/main.c

```diff
@@ -492,12 +494,14 @@ static int suggest_programmers(const char *programmer, LISTID programmers) {
         d[idx].pgmid = ldata(ln2);
         d[idx].desc = pgm->desc;
         d[idx].dist = str_weighted_damerau_levenshtein(d[idx].pgmid, programmer);
+        if(p)
+          d[idx].common_modes = p->prog_modes & pgm->prog_modes;
```

Comment:
If we don't have a valid part, it's better to replace the two lines with
```
        d[idx].common_modes = pgm->prog_modes & (p? p->prog_modes: -1);
 ```
Some programmers don't need the part be specified as they can figure out which part is connected.